src/components/v2/V2Project/V2ProjectSettingsPage/V2ProjectSettings.tsx

```diff
@@ -35,19 +48,99 @@ const items: MenuItem[] = [
 ]
 
 const V2ProjectSettings = () => {
+  const [projectForm] = useForm<ProjectDetailsFormFields>()
+
+  const [loadingSaveChanges, setLoadingSaveChanges] = useState<boolean>()
+  const { projectMetadata, projectId } = useContext(V2ProjectContext)
+
+  const EditV2ProjectDetailsTx = useEditV2ProjectDetailsTx()
+
+  async function onProjectFormSaved() {
+    setLoadingSaveChanges(true)
+
+    const fields = projectForm.getFieldsValue(true)
+
+    const uploadedMetadata = await uploadProjectMetadata({
+      name: fields.name,
+      description: fields.description,
+      logoUri: fields.logoUri,
+      infoUri: fields.infoUri,
+      twitter: fields.twitter,
+      discord: fields.discord,
+      payButton: fields.payButton.substring(0, PROJECT_PAY_CHARACTER_LIMIT), // Enforce limit
+      payDisclosure: fields.payDisclosure,
+    })
+
+    if (!uploadedMetadata.IpfsHash) {
+      setLoadingSaveChanges(false)
+      return
+    }
+
+    EditV2ProjectDetailsTx(
+      { cid: uploadedMetadata.IpfsHash },
+      {
+        onDone: () => setLoadingSaveChanges(false),
+        onConfirmed: async () => {
+          if (projectId) {
+            await revalidateProject({
+              cv: '2',
+              projectId: String(projectId),
+            })
+          }
+          projectForm.resetFields()
+        },
+      },
+    )
+  }
+
+  const resetProjectForm = useCallback(() => {
+    projectForm.setFieldsValue({
+      name: projectMetadata?.name ?? '',
+      infoUri: projectMetadata?.infoUri ?? '',
+      logoUri: projectMetadata?.logoUri ?? '',
+      description: projectMetadata?.description ?? '',
+      twitter: projectMetadata?.twitter ?? '',
+      discord: projectMetadata?.discord ?? '',
+      payButton: projectMetadata?.payButton ?? '',
+      payDisclosure: projectMetadata?.payDisclosure ?? '',
+    })
+  }, [
+    projectMetadata?.name,
+    projectMetadata?.infoUri,
+    projectMetadata?.logoUri,
+    projectMetadata?.description,
+    projectMetadata?.twitter,
+    projectMetadata?.discord,
+    projectMetadata?.payDisclosure,
+    projectMetadata?.payButton,
+    projectForm,
+  ])
+
+  // initially fill form with any existing redux state
+  useEffect(() => {
+    resetProjectForm()
+  }, [resetProjectForm])
+
   return (
     <Layout style={{ minHeight: '100vh', background: 'transparent' }}>
       <Sider style={{ background: 'transparent' }}>
-        <div className="logo" />
+        <Link href={`project`}>{`< Back to Project`}</Link>
```

Comment:
**nitpick** perhaps we could use an antd icon here instead? `<LeftOutlined />` maybe